### DISCUSSION TITLE
relative_time should render texts in local browser language

### DISCUSSION THREAD

**Discussion by BenPru** [home-assistant/frontend#7353] [Localization]
relative_time render texts currently always in english.
Example: "{{ relative_time(states.sun.sun.last_changed) }}" --> "44 minutes" - Current Browser language: german.
So it should be "44 Minuten".

**Comment by bramkragten** [comment]
We get this from the backend, the backend currently doesn't know the user language.

**Reply by BenPru** [reply]
How does it work for the integrated times like last_updated in the frontend?
How can I set the backend language in the docker container?

**Reply by KTibow** [reply]
@BenPru what do you mean for both ones? There is no such thing as a backend language.

**Reply by BenPru** [reply]
"last_updated" is correctly shown in the browser language.
I use ha in a docker container. How can I set the backend language in this docker container?

**Reply by spacegaier** [reply]
Not sure about the Docker container question.

But in general: Instead of having the backend provide `relative_time()` it might be better to provide the `isoformat()` time string because then the frontend can localize the value rendering based on the set HA user language (there are different `format` options for the Lovelace config).

That's is e.g. what is happening with the internal times such as last-changed and last-updated. They provide an ISO datetime string and leave the formatting to the frontend.

**Reply by ElVit** [reply]
Now it's possible to use `state_translated` to get the translated state.
Is it maybe also possible to provide a similar function for `relative_time` e.g. `relative_time_translated` ?

**Reply by Noado** [reply]
Any news here?

**Reply by xrh0905** [reply]
Any ideas on how to fix or mitigate this issue?

**Reply by Noado** [reply]
> Any ideas on how to fix or mitigate this issue?

```
    {% macro humanize_time_ru(seconds) %}{% set minutes = (seconds / 60) |
    round(0, 'floor') %}{% set hours = (seconds / 3600) | round(0, 'floor') %}{%
    if seconds < 60 %}{{ seconds }} {{ 'секунд' if seconds % 10 != 1 or seconds
    == 11 else 'секунду' }}{% elif seconds < 3600 %}{{ minutes }} {{ 'минут' if
    minutes % 10 != 1 or minutes == 11 else 'минуту' }}{% else %}{{ hours }} {{
    'часов' if hours % 10 != 1 or hours == 11 else 'час' }}{% endif %}{%
    endmacro %}
```

I use this macro at the start of text field and this one to generate text itself 

```{{ humanize_time_ru(as_timestamp(now()) - as_timestamp(states.person.noado.last_updated)) }}```

But it's language specific (RU). I've just used DeepSeek to generate it.
